File: Frontend/family-expenses-app/src/types/Expense.ts
export interface BaseExpense {
  // category now references Category by string GUID
  categoryId: string;
  amount: number;
  date: Date;
  description: string;
}

export interface Expense extends BaseExpense {
  // local expenses keep numeric autoincrement id
  id: number;
}

export interface ExpenseFormData extends BaseExpense {
  id?: number;
}

File: Backend/FamilyExpenses.Domain/Entities/Expense.cs
using System.Text.Json.Serialization;

namespace FamilyExpenses.Domain.Entities;

public class Expense
{
    public Guid Id { get; set; }
    public string Description { get; set; } = string.Empty;
    public decimal Amount { get; set; }
    public DateTime Date { get; set; }
    public Guid CategoryId { get; set; }
    
    [JsonIgnore]
    public Category? Category { get; set; }
    
    public Guid FamilyMemberId { get; set; }
    
    [JsonIgnore]
    public FamilyMember? FamilyMember { get; set; }
    
    public string? Notes { get; set; }
    public DateTime LastModified { get; set; }
    public bool IsDeleted { get; set; }
    public string SyncId { get; set; } = string.Empty;
}